converters/fromZigbee.js

```diff
@@ -5662,23 +5662,25 @@ const converters = {
         type: ['attributeReport', 'readResponse'],
         options: [exposes.options.invert_cover()],
         convert: (model, msg, publish, options, meta) => {
-            let running = false;
-
             if (model.model === 'ZNCLDJ12LM' && msg.type === 'attributeReport' && [0, 2].includes(msg.data['presentValue'])) {
                 // Incorrect reports from the device, ignore (re-read by onEvent of ZNCLDJ12LM)
                 // https://github.com/Koenkk/zigbee-herdsman-converters/pull/1427#issuecomment-663862724
                 return;
             }
 
-            if (msg.data['61440']) {
-                const value = msg.data['61440'];
-                // 63025664 comes from https://github.com/Koenkk/zigbee2mqtt/issues/5155#issuecomment-753344248
-                running = value !== 0 && value !== 63025664;
-            }
-
             let position = precisionRound(msg.data['presentValue'], 2);
             position = options.invert_cover ? 100 - position : position;
-            return {position: position, running: running};
+            return {position};
+        },
+    },
+    xiaomi_curtain_direction: {
+        cluster: 'genBasic',
+        type: ['attributeReport', 'readResponse'],
+        convert: (model, msg, publish, options, meta) => {
+            const data = msg.data['1028'] || 0;
+            const lookup = {0: 'STOP', 1: 'FORWARD', 2: 'REVERSE'};
+            const direction = lookup[data] || lookup[0];
+            return {running: direction !== 'STOP', motor_direction: direction};
```

Comment:
We are moving all the `genBasic` reports to a single converter.

You only need to add the `xiaomi_basic` converter to the device:
https://github.com/Koenkk/zigbee-herdsman-converters/blob/4ed602f34946d0678eeb75aa540ac5d0d4ce2491/converters/fromZigbee.js#L5278-L5298

and add your code for the 1028 here:
https://github.com/Koenkk/zigbee-herdsman-converters/blob/4ed602f34946d0678eeb75aa540ac5d0d4ce2491/lib/xiaomi.js#L408-L415

If you put the logs in debug mode you can see if there are other attributes in the message that can be published, in this way the utility method has all the attributes that Xiaomi can send.